### ISSUE TITLE
Add data privacy chapter

### ISSUE THREAD

**Issue by richplant** [alan-turing-institute/the-turing-way#2235]
## Summary

I've been through the current issues and couldn't find anything directly related, but forgive me if I've missed an existing post.

Would it make sense to provide a broad introductory chapter to modern privacy techniques for data science? I'm talking here specifically about the statistical obfuscation of potentially personally identifying data (as defined by the GDPR) in either research datasets that are intended for publication, or in developed models that are intended for public use/dissemination. It seems to me that this is a missing piece of the ethical data science puzzle that the Turing Way doesn't yet address, although I must confess some bias since this is my primary research interest.

This may include some introductory material on various pseduonymity strategies (k-anonymity, l-diversity, and so on), along with more modern material on [differential privacy](https://en.wikipedia.org/wiki/Differential_privacy) and adversarial learning (https://www.research.ed.ac.uk/en/publications/privacy-preserving-neural-representations-of-text).

This section would slot most obviously into the Guide for Ethical Research, although I can certainly imagine it being linked to from the Project Design and Communication guides. 

### Resources

- [Privacy in Deep Learning: A Survey](https://cseweb.ucsd.edu/~fmireshg/survey_pvdl_2020.pdf) *A useful brief overview of some extant threats and mitigation strategies.*
- [Deep learning and differential privacy](https://github.com/frankmcsherry/blog/blob/master/posts/2017-10-27.md) *Frank McSherry's thought provoking blog about the privacy research landscape.*
- [The Algorithmic Foundations
of Differential Privacy](https://privacytools.seas.harvard.edu/files/privacytools/files/the_algorithmic_foundations_of_differential_privacy_0.pdf) *The foundational paper to the study of differential privacy.*

## Updates

- [ ] Write chapter outline
- [ ] Add material to the chapter
- [ ] Combine materials into a readable chapter
- [ ] Proofread
- [ ] Request reviews
- [ ] Address reviews
- [ ] Merge to main branch.

**Comment by Arielle-Bennett** [comment]
Hi @richplant - I think folks like @MariaEriksson were working on some aspects of sensitive data (see https://github.com/alan-turing-institute/the-turing-way/pull/2076) but not sure if this is the detail you were thinking of?

**Comment by richplant** [comment]
Hi @Arielle-Bennett, that seems like an interesting resource, but it's a little more narrowly construed than what I had in mind - I'm suggesting some material about privacy risks specifically arising from aggregating non-uniquely identifying data with secondary sources, which isn't solely concerned with sharing datasets online. I'm planning to get an outline together shortly, perhaps that will spell out the differences more clearly.

**Comment by MariaEriksson** [comment]
Hey @richplant! Thank you for your suggestion. @EKaroune and I are currently expanding the Sensitive data/projects chapter, which covers the following sections (that have already been written by others): 

- Background about sensitive data by @EstherPlomp
- Tips for working openly with sensitive data by myself
- Collaborating on a sensitive data project by @EKaroune
- Data safe havens by @edwardchalstrey1

Here are some of our [notes](https://hackmd.io/DRiGvOgbRdy9YkYdUzZC7A). This is very much a work in progress. If you feel like you can add to this, please do. We are also looking for case studies, and it sounds like what you are suggesting might lend itself nicely as such?

**Comment by richplant** [comment]
Hi @MariaEriksson, it looks like our sections could be compatible - I certainly would appreciate not having to re-cover all the ground you propose in the overview! 

My suggested chapter applies quite directly to some of the distinctive problems of machine learning and mitigating them technically - perhaps that could be added as part of your 'combination of different datasets' section. The remedies I'm suggesting would also fit in nicely alongside your 'Managing sensitive data projects'.

In terms of case studies, I work with the Data Study Group team, we regularly manage sensitive data of various kinds during our projects - I can think of a few instances that might yield interesting stories, if you're interested?

**Comment by MariaEriksson** [comment]
Thanks @richplant! That sounds like a great plan. @EKaroune and I will be adding some bits to the overview tomorrow during the Turing Way Collaboration Café - you’re welcome to come along too. Otherwise, just add to it when you feel able to 👍 

I think the case studies from the DSG team would be very interesting. Would you be able to provide a short description of the projects you think would best lend themselves to the chapter?

**Comment by richplant** [comment]
Cheers Maria, I'm working to some paper deadlines right now, but I'll add a skeleton for the proposed sections as soon as I can. Do you have a branch/fork you prefer to work on? 

There are a few DSGs that spring to mind - I can't share any non-public details without clearing them, but here are a couple that leveraged sensitive input data, so demonstrate pseudonymisation, secure environments, publishing derived information, and so on: 

* [Bristol University Heart Disease](https://www.turing.ac.uk/research/publications/data-study-group-final-report-bristol-university-applying-ai-and-machine)
* [Department for Work and Pensions](https://www.turing.ac.uk/research/publications/data-study-group-final-report-department-work-and-pensions)

We also have a few others in review in the medical domain that would fit the bill.

**Comment by likeajumprope** [comment]
Hi @richplant I was wondering whether I could work on this during the next book dash - maybe we can get in touch?

**Comment by richplant** [comment]
Absolutely, I'd love to collaborate, you can email r.plant@napier.ac.uk

**Comment by EKaroune** [comment]
Thanks for your reply @richplant - it's best to keep conversations in Github and then they are recorded here. It would be great for you to share anything you have already written with @likeajumprope and she can add to that during the bookdash. You could then review it. 

If you want to come to a co-working session before or during the bookdash to discuss this more, we could arrange that.

**Comment by richplant** [comment]
@EKaroune I'll add what I've got to the PR, and I'd be happy to discuss during a co-working session.

**Comment by likeajumprope** [comment]
> Thanks for your reply @richplant - it's best to keep conversations in Github and then they are recorded here. It would be great for you to share anything you have already written with @likeajumprope and she can add to that during the bookdash. You could then review it.
> 
> If you want to come to a co-working session before or during the bookdash to discuss this more, we could arrange that.

Hi @richplant and @EKaroune thanks for getting me on board! It would be perfect if you could share what you have already :) ! And I would love to come to a co-working session before the dash (fingers crossed that my application gets through) :)

**Comment by richplant** [comment]
@likeajumprope @EKaroune @MariaEriksson  I've raised a PR against the existing branch (#2339), so I'm happy to move discussion over to there and close this PR.

In regards to the case studies, perhaps I could add a brief discussion of the DSG method plus some examples from recent groups (we'll be publishing a number of new reports soon) once the section on collaboration is up?

**Comment by EKaroune** [comment]
That sounds fabulous! I'm going to try to worm on the sensitive data chapter overview  next week so that we can start to structure the chapters in the github repo during the bookdash and do final edits. Then we can add in any other bits of writing. Thanks @richplant
